custom/intrahealth/sqldata.py

```diff
@@ -5609,17 +5609,28 @@ def aggregated_data(self):
             'Net à Payer': 0,
         }
         valid_products = self.program_products
+        delivery_amt_owed_dict = {}
         for row in rows:
+            pps_name = row['pps_name']
             if valid_products and \
                row['product_id'] not in valid_products:
                 continue
 
             data['Total Facture'] += self.get_value(row['amount_billed'])
-            data['Net à Payer'] += self.get_value(row['amount_owed'])
+            delivery_amt_owed = self.get_value(row['delivery_amt_owed'])
+            if delivery_amt_owed_dict.get(pps_name, None):
+                delivery_amt_owed_dict[pps_name].add(delivery_amt_owed)
+            else:
+                delivery_amt_owed_dict[pps_name] = set()
+                delivery_amt_owed_dict[pps_name].add(delivery_amt_owed)
 
         # data['Net à Payer'] = int(data['Total Facture'] * 1.075)
         data['Total Facture'] = round(float(data['Total Facture']), 2)
-        data['Net à Payer'] = round(float(data['Net à Payer']), 2)
+        net_a_payer = 0
+        for pps_values in delivery_amt_owed_dict.values():
+            for value in pps_values:
+                net_a_payer += value
+        data['Net à Payer'] = round(float(net_a_payer), 2)
```

Comment:
This still doesn't seem to make sense.  What are you trying to do?  Is this supposed to be a simple total of all `delivery_amt_owed` values?  Why not just add them up?

Reply:
@esoergel This report is using multiple visits for it's calculations, without `set()` values building 'Net à Payer' are recurring for each of ppses and 'Net à Payer' is too high.

Reply:
It sounds like the values in `row['delivery_amt_owed']` are repeated, then, is that right?  Does each PPS have multiple deliveries as well?

It sounds like this should only count the amount owed once per delivery.  This instead counts the amount owed when the value is unique for that PPS.  What if there are multiple deliveries to the same PPS for the same value?  For instance, I could imagine there being 3 rows for a particular PPS with a `delivery_amt_owed` of $10.  Even if the second row represents the same delivery as the first, the third could be a second delivery.  With this algorithm, the second delivery wouldn't be counted.